### PR TITLE
chore(Breadcrumb): make small screen BreadcrumbMultiple collapse when resizing to medium screen

### PR DESCRIPTION
**Pull request by joakbjerk**
Before:

https://github.com/dnbexperience/eufemia/assets/25927156/4f33a2ee-784f-413f-aa9a-73648998df76

After:

https://github.com/dnbexperience/eufemia/assets/25927156/8cc71bdd-6103-4c2a-abcc-41d3355906c9

No tests at the moment, as testing was made difficult due to all the elements having a width of 0 in the jest-dom

- [x] Add test
- [x] Double check SSR issues

using the `useMediaQuery` to only collapse the `BreadcrumbMultiple` on screen change does not seem to cause hydration issues.

### CODE REVIEW

File: packages/dnb-eufemia/src/components/breadcrumb/__tests__/Breadcrumb.test.tsx [hunk 1]
```diff
@@ -271,6 +274,42 @@ describe('Breadcrumb', () => {
...
+
+    const toggleButton = () =>
+      document.querySelector('.dnb-breadcrumb__toggle')
+
+    const collapseSection = document.querySelector(
+      '.dnb-breadcrumb__collapse'
+    )
+
+    // Collapsable menu should not be visible before toggle click
+    expect(collapseSection.children.length).toBe(0)
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/breadcrumb/__tests__/Breadcrumb.test.tsx:298]:
```suggestion
    expect(collapseSection.children).toHaveLength(0)
```

File: packages/dnb-eufemia/src/components/breadcrumb/__tests__/Breadcrumb.test.tsx [hunk 2]
```diff
@@ -271,6 +274,42 @@ describe('Breadcrumb', () => {
...
+
+    await userEvent.click(toggleButton())
+
+    // Collapsable should be visible now
+    expect(collapseSection.children.length).toBe(1)
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/breadcrumb/__tests__/Breadcrumb.test.tsx:303]:
```suggestion
    expect(collapseSection.children).toHaveLength(1)
```

File: packages/dnb-eufemia/src/components/breadcrumb/__tests__/Breadcrumb.test.tsx [hunk 3]
```diff
@@ -271,6 +274,42 @@ describe('Breadcrumb', () => {
...
+
+    act(() => {
+      setMedia({ width: '80em' })
+    })
+
+    // Collapsable menu should auto-close when screen goes large
+    expect(collapseSection.children.length).toBe(0)
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/breadcrumb/__tests__/Breadcrumb.test.tsx:310]:
```suggestion
    expect(collapseSection.children).toHaveLength(0)
```

**Review by tujoworker — APPROVED** [review]
Nice! ⭐

**Comment by tujoworker** [comment]
Could you verify that this works on SSR too?

**Comment by joakbjerk** [comment]
> Could you verify that this works on SSR too?

Seems to be working after testing in the portal

**Pull request merged**